lib/wasi/src/utils.rs

```diff
@@ -1,18 +1,37 @@
 use wasmer_runtime_core::module::Module;
 
+#[allow(dead_code)]
 /// Check if a provided module is compiled with WASI support
 pub fn is_wasi_module(module: &Module) -> bool {
-    if module.info().imported_functions.is_empty() {
-        return false;
-    }
-    for (_, import_name) in &module.info().imported_functions {
-        let namespace = module
-            .info()
-            .namespace_table
-            .get(import_name.namespace_index);
-        if namespace != "wasi_unstable" {
-            return false;
+    get_wasi_version(module) == Some(WasiVersion::Snapshot1)
+}
+
+/// The version of WASI.  This is determined by the namespace string
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum WasiVersion {
+    /// "wasi_unstable"
+    Snapshot0,
+    /// "wasi_snapshot_preview1"
+    Snapshot1,
+}
+
+pub fn get_wasi_version(module: &Module) -> Option<WasiVersion> {
+    let mut import_iter = module
+        .info()
+        .imported_functions
+        .iter()
+        .map(|(_, import_name)| import_name.namespace_index);
+
+    // returns None if empty
+    let first = import_iter.next()?;
+    if import_iter.all(|idx| idx == first) {
```

Comment:
What is this `.all(|idx| idx == first)` doing? 

Because Emscripten implements part of `wasi` (setting some of the imported functions  as `wasi_unstable`), we need to check that all the imports have this namespace (not just one). That way we can differentiate Emscripten and WASI modules

Reply:
It checks that all namespaces have the same index, which I understood to be a faster way to check if they all share a namespace than looking up every one and doing string comparisons.

After it guarantees that they're all the same, it does logic on it.  I can add comments to clarify this logic

Reply:
Actually, it creates a bug. If the import namespaces are the following: `wasi_unstable` and `env`, we have 2 different namespaces, and `get_wasi_version` will return `None`. Working on a fix.

Reply:
@Hywan as mentioned in #1028 , I think this is not a bug and that the behavior in #1028 is much closer to a bug because we'll start calling things which are not WASI (won't run with WASI imports) WASI and have no way to detect real WASI anymore.

Reply:
Is it specified in the specification that, in case of WASI, all imports must land in the same namespace? I think it is very restrictive, and would be break some design (like blockchain or stuff like that that have special namespaces too).